mlir/lib/Dialect/MQTOpt/Transforms/ToQuantumComputationPattern.cpp

```diff
@@ -266,52 +316,55 @@ struct ToQuantumComputationPattern final : mlir::OpRewritePattern<AllocOp> {
     circuit.addQubitRegister(numQubits, regName);
     circuit.addClassicalRegister(numQubits);
 
-    std::unordered_set<mlir::Operation*> visited{};
+    std::set<mlir::Operation*> visited{};
+    std::set<mlir::Operation*> mqtUsers{};
```

Comment:
I am fairly certain that these should both be `unordered_set` as you want to use them as hash maps with quick membership checks and you do not need these to be sorted.